vscode/src/extension.ts

```diff
@@ -103,6 +103,21 @@ function registerDocumentUpdateHandlers(languageService: ILanguageService) {
     updateIfQsharpDocument(document);
   });
 
+  // we manually send an OpenDocument telemetry event if this is a Q# document, because the
+  // below subscriptions won't fire for documents that are already open when the extension is activated
+  vscode.workspace.textDocuments.forEach((document) => {
+    if (isQsharpDocument(document)) {
+      const documentType = isQsharpNotebookCell(document)
```

Comment:
So if I open a notebook with a couple dozen Q# cells, is this going to send a couple dozen telemetry events? Or do we consider opened a notebook one document?

Reply:
This will get fired for each individual Q# cell. 

That being said I personally am not concerned about over-firing here.